### PR TITLE
fix up python version to 3 and add apk upgrade

### PR DESCRIPTION
**Pull request by williamfalconeruk**
## Purpose
_Briefly describe the purpose of the change, and/or link to the JIRA ticket for context_

To fix an issue with Zap DockerFile that was causing issues locally when spinning up in docker-compose. This does not commission the docker file to run the zap scripts it has yet.

## Approach
- Upgrade PIP to PIP3 and add some prerequisites and no-cache to prevent warnings.
- add an Apk upgrade to prevent vuln creep.
- Made the docker file clearer in terms of steps taken by running each step separately.

## Learning

- general research around docker and python:
https://stackoverflow.com/questions/7446187/no-module-named-pkg-resources 
https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management#Upgrade_a_Running_System
 - also, what is zap? 
https://www.zaproxy.org/

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: zap/Dockerfile [hunk 1]
```diff
@@ -4,8 +4,12 @@ WORKDIR /zap
 
 USER root
 
-RUN apk add python3 py-pip && rm -rf /var/cache/apk/* \
-    && pip install --upgrade zapcli
+RUN apk upgrade
```
Comment (cooperaj) [inline zap/Dockerfile]: Splitting up a run command makes it clearer but it actually does bad juju to the resultant images. Each command in a docker file makes a layer that is always a part of the image. If you run the upgrade and install and *then* run a cache clear up your image will contain the cache files twice and then be removed.

File: zap/Dockerfile [hunk 2]
```diff
@@ -4,8 +4,12 @@ WORKDIR /zap
...
+RUN  apk add python3 py3-pip
+RUN rm -rf /var/cache/apk/*
+RUN pip3 install --no-cache-dir --upgrade pip
```
Comment (cooperaj) [inline zap/Dockerfile]: The install of pip and setup tools is not needed as they're bundled as a part of py3-pip system package

File: zap/Dockerfile (hunk 1 above)
Comment (williamfalconeruk) [reply zap/Dockerfile]: ah yes, makes a bit of sense now. will revert to doing that

File: zap/Dockerfile (hunk 2 above)
Comment (williamfalconeruk) [reply zap/Dockerfile]: ok - will remove. I apologize for not understanding this.

File: zap/Dockerfile [hunk 3]
```diff
@@ -4,12 +4,10 @@ WORKDIR /zap
...
-RUN apk upgrade
-RUN  apk add python3 py3-pip
-RUN rm -rf /var/cache/apk/*
-RUN pip3 install --no-cache-dir --upgrade pip
-RUN pip3 install --no-cache-dir --upgrade setuptools
-RUN pip3 install --no-cache-dir --upgrade zapcli
+RUN apk upgrade \
+ && apk add python3 py3-pip \
+ && rm -rf /var/cache/apk/* \
+ && pip3 install --no-cache-dir --upgrade pip zapcli
```
Comment (williamfalconeruk) [inline zap/Dockerfile]: Note pip has a version 20 which is why it is here (the base one comes down as 18)

Comment (cooperaj) [reply zap/Dockerfile]: Yeah, I don't see that being an issue. The OS packages will contain security fixes and having the latest version of PIP isn't necessary unless the installation of zapcli requires it.

Comment (williamfalconeruk) [reply zap/Dockerfile]: @cooperaj - ready for another review - let me know if ok with this.

Comment (williamfalconeruk) [reply zap/Dockerfile]: I was trying to get rid of the warnings that came up. but happy either way :)

File: zap/Dockerfile [hunk 4]
```diff
@@ -7,7 +7,7 @@ USER root
 RUN apk upgrade \
  && apk add python3 py3-pip \
  && rm -rf /var/cache/apk/* \
- && pip3 install --no-cache-dir --upgrade pip zapcli
+ && pip3 install --no-cache-dir --upgrade zapcli
```
Comment (williamfalconeruk) [inline zap/Dockerfile]: Should be good to go now.

**Pull request merged**